assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js

```diff
@@ -88,11 +95,38 @@ function PageLayouts(props) {
     overscan: 2,
   });
 
+  const columnVirtualizer = useVirtual({
+    horizontal: true,
+    size: 2,
+    parentRef,
+    estimateSize: useCallback(() => pageSize.width + PAGE_LAYOUT_ROW_GAP, [
+      pageSize.width,
+    ]),
+    overscan: 0,
+  });
+
   const requiresConfirmation = useMemo(
     () => currentPage && !isDefaultPage(currentPage),
     [currentPage]
   );
 
+  const onTabKeyDown = useCallback(() => {
+    // When page layouts is already in focus and tab is hit again we want to move focus out of the panel
+    // This gets a little goofy because there's nothing else focusable left to go to in the page layouts tab
+    // and if we back up without moving to the next sibling we'll never let the keyboard user continue through the page.
+    if (parentRef.current) {
+      const canvas =
+        parentRef.current?.offsetParent?.offsetParent?.offsetParent?.nextSibling
+          ?.firstChild;
```

Comment:
How often does the layout change? Will the canvas always be locatable in that position?

Maybe we could add a test for this in the event that the canvas isn't locatable through that ref. Or we could try to pass in the canvas ref but that could be annoying 🤔

Reply:
soooo I'm hoping that there's a better solution to getting the canvas to focus. But you do bring up good questions. 

1, The canvas will always be this next sibling from the panel, the dom structure is a parent section and it has 3 children: the initial set of panels, the canvas, then the other set of panels. So we can reliably say that the canvas will always be that middle child and therefore the next sibling.  
2. Passing the canvas ref isn't going to be possible because of how the code base is separated. Ideally the focus gets handed off to the next available focusable element in the dom, but since page layouts has nothing after it (either as a next panel or content in the same panel) it's just a barren waste land of nothingness to take focus. The panel ref that I have access to is unaware of anything else to focus. We could add a hidden button to take focus below the page layouts however that's going to do weird things to the scroll and be confusing for anyone using a screen reader.

Reply:
I think it's reasonable to assume the layout will not change often so it is a valid selector, but I agree that an integration test would be good so that we can be assured the functionality is there if anything does change.

Otherwise, it seems to me like we would be adding the canvas ref to a context accessible to the panel, which seems like it could also be fragile.

Reply:
yeah, I'm gonna at Morten to review to make sure I didn't miss something that's available to tab out of the panel - if there isn't a better way to do this then I'll add a test.